File: BACKEND/facturador_backend/src/main/java/com/bap/api/services/api/ApiSincronizacionService.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.bap.api.services.api;

import com.bap.api.model.api.ApiSincronizacion;
import com.bap.api.dto.RespuestaSincronizacion;
import java.util.List;
import com.bap.api.configurate.AbstractEntity;
import com.bap.api.dto.SolicitudCliente;

/**
 *
 * @author ruth
 *
 */
public interface ApiSincronizacionService extends AbstractEntity<ApiSincronizacion> {

    List<RespuestaSincronizacion> sincronizacionDiaria(String login);

    List<RespuestaSincronizacion> codigosCatalogos(String grupo);

    RespuestaSincronizacion sincronizacionCodigoAutorizacion(String grupo, String login, int codigoAutorizacion);

    RespuestaSincronizacion solicitudNuevoProducto(SolicitudCliente t);

    RespuestaSincronizacion validacionSolicitudNuevoProducto(SolicitudCliente t);

}

File: FRONTEND/facturador-frontend/src/services/ApiSincronizacionService.js
import Api from '@/services/Api'

export default {
    ///ApiSincronizacion
    getDatosSincronizados(grupo){
        return Api().get("/api/apiSincronizacion/codigoCatalogos", { params: { grupo: grupo }})
    },
    sincronizacionPorSolicitud(grupo,login, codigoAutorizacion){
        return Api().get(`/api/apiSincronizacion/sincronizacionBySolicitud/${grupo}/${login}/${codigoAutorizacion}`)
    },
    solicitudNuevoProducto(dato){
        return Api().post("/api/apiSincronizacion/solicitaNuevoProducto", dato);
    },
    validaSolicitudNuevoProducto(dato){
        return Api().post("/api/apiSincronizacion/validacionSolicitudNuevoProducto", dato);
    },
    sincronizacionDiaria(login){
        return Api().get(`/api/apiSincronizacion/sincronizacionDiaria/${login}`)
    },
}
